misc/random.c

```diff
@@ -44,6 +47,12 @@ void mp_rand_seed(uint64_t seed)
 #ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
     seed = 42;
 #endif
+
+    if (seed == 0) {
+        if (av_random_bytes((uint8_t *)&seed, sizeof(seed)) < 0)
```

Comment:
This appears to be correct and we would have to use an union.
no wonder that C code is so buggy given such obscure rules...

Reply:
Or use standard approved way is to do `memcpy` in such cases.

``` c
    char buf[sizeof(seed)];
    if (av_random_bytes((uint8_t *)&buf, sizeof(buf)) < 0) {
        seed = mp_raw_time_ns();
    } else {
        memcpy(&seed, buf, sizeof(seed));
    }
```

though of course in practice it doesn't really matter...

Reply:
That's exactly how libavutil uses it, though. :-} (Although with uint32_t instead of uint64_t)

I honestly have no idea how to fix that. I don't use C enough for the finer details…

I could do a revision of [my original idea](https://github.com/frostschutz/mpv/tree/random-seed-v1) with time, pid, `getentropy()`, or even just skip the getentropy part altogether. The pid alone would pretty much fix my use-case. I did the extra step because it seemed right and simple to do so, but nothing's ever simple, is it…

At the end of the day it's just for shuffling playlists, not aes keys. Time is good enough for most people already.

What do you think?

Reply:
Don't worry, it is silly issue, that is not that important. Let's wait for @na-na-hi to comment what would be their prefered solution here. I would opt for memcpy, but let's sync up with everyone.

Reply:
> That's exactly how libavutil uses it, though. :-}

They have a (potential) bug then.

> At the end of the day it's just for shuffling playlists, not aes keys.

Unfortunately if you do UB in C it can also mean that the compiler decides the seed is always `1234`.
Then even your playlist shuffling wouldn't work.